test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs

```diff
@@ -301,11 +301,14 @@ public void ConnectionCanReadAndWrite(TestServiceContext testContext)
             {
                 using (var connection = new TestConnection(server.Port))
                 {
-                    await connection.Send(
```

Comment:
Maybe we should add a ChunkedRequestsTests class much like the ChunkedResponseTests one. EngineTests is getting kinda large. We could add a test method for trailing headers while we're at it.